app/src/main/java/br/org/cesar/discordtime/stickysessions/domain/interactor/SaveCurrentUser.java

```diff
@@ -0,0 +1,30 @@
+package br.org.cesar.discordtime.stickysessions.domain.interactor;
+
+import br.org.cesar.discordtime.stickysessions.domain.repository.UserRepository;
+import io.reactivex.Single;
+import io.reactivex.SingleObserver;
+
+public class SaveCurrentUser extends UseCase<String, Boolean> {
+
+    private UserRepository mUserRepository;
+
+    public SaveCurrentUser(UserRepository userRepository) {
+        mUserRepository = userRepository;
+    }
+
+    @Override
+    public Single<Boolean> execute(String userName) {
+
+        if (userName == null || userName.trim().isEmpty()) {
+            return Single.error(new Exception("The user is invalid (null or empty)"));
+        }
+
+        boolean userSaved = mUserRepository.saveUser(userName);
+        return new Single<Boolean>() {
```

Comment:
Since `userSaved` is just a boolean that you already have the value, you can refactor this code to use: `Single.just(userSaved)`.

Reference: http://reactivex.io/documentation/single.html